rcljava/src/main/java/org/ros2/rcljava/action/ActionServerImpl.java

```diff
@@ -432,6 +486,93 @@ private static native boolean nativeCheckGoalExists(
     long goalInfoFromJavaConverterHandle,
     long goalInfoDestructorHandle);
 
+  private boolean goalExists(action_msgs.msg.GoalInfo goalInfo) {
+    long goalInfoFromJavaConverterHandle = goalInfo.getFromJavaConverterInstance();
+    long goalInfoDestructorHandle = goalInfo.getDestructorInstance();
+
+    return nativeCheckGoalExists(
+      this.handle, goalInfo, goalInfoFromJavaConverterHandle, goalInfoDestructorHandle);
+  }
+
+  private static native void nativePublishStatus(long handle);
+
+  private void publishStatus() {
+    this.nativePublishStatus(this.handle);
+  }
+
+  private static native void nativePublishFeedbackMessage(
+    long handle,
+    FeedbackMessageDefinition feedbackMessage,
+    long feedbackMessageFromJavaConverterHandle,
+    long feedbackMessageDestructorHandle);
+
+  private void publishFeedbackMessage(FeedbackMessageDefinition<T> feedbackMessage) {
+    this.nativePublishFeedbackMessage(
+      this.handle,
+      feedbackMessage,
+      feedbackMessage.getFromJavaConverterInstance(),
+      feedbackMessage.getDestructorInstance());
+  }
+
+  private static native void nativeNotifyGoalDone(long handle);
+
+  private void notifyGoalDone() {
+    this.nativeNotifyGoalDone(this.handle);
+  }
+
+  private static native void nativeExpiredGoals(
+    long handle, action_msgs.msg.GoalInfo goalInfo,
+    long goalInfoToJavaConverterHandle, Consumer<action_msgs.msg.GoalInfo> onExpiredGoal);
+
+  private void expiredGoals() {
+    action_msgs.msg.GoalInfo goalInfo = new action_msgs.msg.GoalInfo();
+    long goalInfoToJavaConverterHandle = goalInfo.getFromJavaConverterInstance();
+    nativeExpiredGoals(
+      this.handle, goalInfo, goalInfoToJavaConverterHandle,
+      new Consumer<action_msgs.msg.GoalInfo>() {
+        public void accept(action_msgs.msg.GoalInfo goalInfo) {
+          List<Byte> goalUuid = goalInfo.getGoalId().getUuidAsList();
+          ActionServerImpl.this.goalResults.remove(goalUuid);
+          ActionServerImpl.this.goalRequests.remove(goalUuid);
+          ActionServerImpl.this.goalHandles.remove(goalUuid);
+        }
+      });
+  }
+
+  private action_msgs.msg.GoalInfo createGoalInfo(List<Byte> goalUuid) {
+    action_msgs.msg.GoalInfo goalInfo = new action_msgs.msg.GoalInfo();
+    unique_identifier_msgs.msg.UUID uuidMessage= new unique_identifier_msgs.msg.UUID();
+    uuidMessage.setUuid(goalUuid);
+    goalInfo.setGoalId(uuidMessage);
+    return goalInfo;
+  }
+
+  private ResultResponseDefinition<T> createResultResponse() {
+    ResultResponseDefinition<T> resultResponse;
+    try {
+      resultResponse = this.actionTypeInstance.getGetResultResponseType().newInstance();
+    } catch (Exception ex) {
+      throw new IllegalArgumentException("Failed to instantiate provided action type", ex);
+    }
+    return resultResponse;
+  }
+
+  private void publishResult(List<Byte> goalUuid, ResultResponseDefinition<T> resultResponse) {
```

Comment:
Maybe rename this to "sendResult" for consistency, since it ultimately is a ROS service (not a publisher).

I guess it is a little confusing to also have another method named `sendResultResponse`. Adding some docs might help clarify the differences.